### PR TITLE
prediction with audio file on honking

### PR DESCRIPTION
**Pull request by ljj7975**
training was returning incorrect accuracy and high loss due to the way I use `categoricalCrossentropy` loss function. When model is compiled with `categoricalCrossentropy`, it expects soft_max layer as a last layer of the model.

After fixing above issue, I was able to get 90% > accuracy for 4 commands (go, no, yes, stop).

Updated honking side code to load the new weights and predicts input audio files.
result seems reasonable

### CODE REVIEW

File: training/training.py [hunk 1]
```diff
@@ -181,22 +185,27 @@ def prepare_dataset(command_list, data_dir, input_shape):
...
-
-        original_data = np.loadtxt(data_dir + data_file)
-        data = np.pad(original_data, (0, 4000 - len(original_data)), 'constant').reshape(input_shape)
-        X.append(data)
-        Y.append(index)
+    for subdir, dirs, files in os.walk(data_dir):
+        for file in files:
+            if file.startswith('.'):
+                continue
+            target_command = "unkonwn"
```
Comment (daemon) [inline training/training.py]: `unkonwn` is a typo?

File: audio.js [hunk 2]
```diff
@@ -186,6 +186,9 @@ class Audio {
   };
 
   get_data() {
+    while (this.mfcc_flattened.length < 4000) {
+      this.mfcc_flattened.push(0);
```
Comment (daemon) [inline audio.js]: Minor comment: maybe use camelCase for JS?

File: training/training.py [hunk 3]
```diff
@@ -181,22 +185,27 @@ def prepare_dataset(command_list, data_dir, input_shape):
...
+            if target_command == "unknown":
+                continue;
+
+            original_data = np.loadtxt(os.path.join(subdir, file))
+	    if len(original_data) < 4000:
+                total_pad = 4000 - len(original_data);
+	        left = np.random.randint(total_pad);
+                data = np.pad(original_data, (left, total_pad-left), 'constant')
+            data = data.reshape(input_shape)
+	    X.append(data)
```
Comment (daemon) [inline training/training.py]: Another minor comment: tabs and spaces being mixed here.

**Comment by ljj7975** [comment]
mic input has been linked to model prediction

**Pull request merged**